nx_arangodb/classes/graph.py

```diff
@@ -355,17 +347,20 @@ def __set_db(self, db: Any = None) -> None:
                 m = "arango.database.StandardDatabase"
                 raise TypeError(m)
 
-            db.version()
+            db.version()  # make sure the connection is valid
             self.__db = db
+            self._db_name = db.name
+            self._hosts = db._conn._hosts
+            self._username, self._password = db._conn._auth
```

Comment:
Setting environment variables (i.e `DATABASE_HOST`, `DATABASE_USER`, etc.) is still supported, but passing a custom `db` object will take priority over env vars